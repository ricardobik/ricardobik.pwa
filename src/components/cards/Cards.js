import React, { Component } from 'react';

import CardItem from '../card-item/CardItem';
import styles from './styles.scss';

class Cards extends Component {
  constructor(props) {
    super(props);

    this.state = {
      openedCard: null,
    };

    this.cards = [
      '?', '0', '1/2', '1', '2', '3', '5', '8', '13', '21', '34', '55',
    ];

    this.toggleCard = this.toggleCard.bind(this);
    this.renderCards = this.renderCards.bind(this);
  }

  toggleCard(card) {
    this.setState({ openedCard: card });
  }

  renderCards() {
    if (this.state.openedCard) {
      return (
        <div className={styles.card}>
          <CardItem
            toggleCard={this.toggleCard}
            value={this.state.openedCard}
            openedCard={this.state.openedCard}
          />
          <div className={styles.quotes}>
            <p className={styles.firstText}>If there is no struggle,</p>
            <p>there is no progress.</p>
          </div>
        </div>
      );
    }
    return this.cards.map(card =>
      <CardItem
        key={card}
        toggleCard={this.toggleCard}
        value={card}
        openedCard={this.state.openedCard}
      />,
    );
  }

  render() {
    const animationStyles = this.state.openedCard ?
      styles.opened : styles.closed;

    return (
      <div className={styles.container}>
        <div className={styles.content}>
          <div className={animationStyles}>
            {this.renderCards()}
          </div>
        </div>
      </div>
    );
  }
}

export default Cards;
