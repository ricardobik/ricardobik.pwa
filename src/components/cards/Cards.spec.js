import React from 'react';
import renderer from 'react-test-renderer';

import Cards from './Cards';

test('Snapshot', () => {
  const component = renderer.create(
    <Cards />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
