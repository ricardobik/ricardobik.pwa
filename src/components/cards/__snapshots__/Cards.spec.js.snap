// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<div
  className="container"
>
  <div
    className="content"
  >
    <div
      className="closed"
    >
      <div
        className="container"
        onClick={[Function]}
      >
        <span>
          ?
        </span>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <span>
          0
        </span>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <span>
          1/2
        </span>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <span>
          1
        </span>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <span>
          2
        </span>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <span>
          3
        </span>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <span>
          5
        </span>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <span>
          8
        </span>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <span>
          13
        </span>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <span>
          21
        </span>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <span>
          34
        </span>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <span>
          55
        </span>
      </div>
    </div>
  </div>
</div>
`;
